### PR TITLE
Bug 1688178 - Big Sur/M1 updates: xcframeworks, using Xcode clang, lipo

### PR DESCRIPTION
**Pull request by badboy**
This is my first try at building Glean iOS on a M1 (arm) machine.

It essentially requires us to use xcframeworks, because the Carthage workaround is not suitable on M1 machines (as we can't exclude the arm64 target, because that IS the host target).
I can build Glean iOS now on my M1.
I can't run the tests though, because [Rust doesn't know the difference between the host system and the simulator](https://github.com/rust-lang/rust/issues/81632).

It's also likely that our sample app won't work the same anymore, at least not if I also want to get that compilable and runnable on an M1. It will need to start using xcframeworks too.
Carthage currently cannot produce xcframework binary releases, so the sample will always need to compile Glean.
Consumers wouldn't be affected, as long as they don't require xcframeworks. It just means I can't build those projects.

* [x] Test it on an Intel Mac
* [x] Ensure tests on CI work
* [x] Ensure sample app on CI works
* [x] Ensure our release still works for consumers
	* Actually probably not as important given that the only active consumer (firefox-ios) gets it through application-services.
* [x] Ensure that application-services doesn't break

### CODE REVIEW

**Review by travis79 — APPROVED** [review]
havewebeenappledyet.com?   LGTM!

File: .circleci/config.yml [hunk 1]
```diff
@@ -667,18 +661,14 @@ jobs:
...
             CARTFILE_PATH="${GLEAN_PATH}/samples/ios/app/Cartfile"
-            # The previous step generated a binary file and the corresponding JSON manifest
-            JSON_URL="https://circleci.com/api/v1.1/project/github/mozilla/glean/${IOS_BUILD_NUM}/artifacts/0/dist/mozilla.glean.json"
 
             echo "Current Cartfile:"
             cat "${CARTFILE_PATH}"
             echo "================="
             echo "New Cartfile:"
-            sed -i.bak "/mozilla\/glean/s#.*#binary \"$JSON_URL\" ~> 0.0.1-SNAPSHOT#" "$CARTFILE_PATH"
+            sed -i.bak "/mozilla\/glean/s#.*#github \"mozilla/glean\" \"${CIRCLE_SHA1}\"#" "$CARTFILE_PATH"
```
Comment (travis79) [inline .circleci/config.yml:671]: Eliminating the "snapshot" really cleaned up this section of the script

**Review by travis79 — APPROVED** [review]
Still looks good to me!

File: .circleci/config.yml (hunk 1 above)
Comment (badboy) [reply .circleci/config.yml:671]: ... but also slower :(

**Comment by badboy** [comment]
Now that the 2 iOS tasks don't depend on each other we can run them in parallel! The summed up runtime will be the same, but the actual runtime should be shorter!

**Pull request merged**